src/Integration.UnitTests/Suppression/SonarQubeIssuesProviderTests.cs

```diff
@@ -205,95 +205,250 @@ public void GetIssues_NoIssuesOnServer_ReturnsEmptyList()
-            var issue2 = new SonarQubeIssue("folder1/file1", "hash2", 0, "message", "sqkey:sqkey:projectID2",
-                SonarQubeIssueResolutionState.WontFix, "S102");
-            var issue3 = new SonarQubeIssue("folder1/file1", "hash3", 0, "message", "sqkey:sqkey:projectID2",
-                SonarQubeIssueResolutionState.Fixed, "S103");
-            var issue4 = new SonarQubeIssue("folder1/file1", "hash4", 0, "message", "sqkey:sqkey: projectID2",
-                SonarQubeIssueResolutionState.WontFix, "S104");
-            var issue5 = new SonarQubeIssue("folder1/file1", "hash5", 0, "message", "sqkey:XXX:projectID2",
-                SonarQubeIssueResolutionState.FalsePositive, "S105");
-
-            var issue6 = new SonarQubeIssue("folder1/file2", "hash6", 0, "message", "sqkey:sqkey:projectID1",
-                SonarQubeIssueResolutionState.Unresolved, "S106");
-            var issue7 = new SonarQubeIssue("folder2/file1", "hash7", 0, "message", "sqkey:sqkey:projectID1",
-                SonarQubeIssueResolutionState.Fixed, "S107");
-
-            var issue8 = new SonarQubeIssue("file1", "hash8", 0, "message", "sqkey:sqkey:projectID1",
-                SonarQubeIssueResolutionState.FalsePositive, "S108");
-            var issue9 = new SonarQubeIssue("file1", "hash9", 0, "message", "sqkey:sqkey:projectID1xxx",
-                SonarQubeIssueResolutionState.WontFix, "S109");
-            var issue10 = new SonarQubeIssue("file1", "hash10", 0, "message", "sqkey:sqkey:projectID",
-                SonarQubeIssueResolutionState.Unresolved, "S110");
-
-            SetupSolutionBinding(isConnected: true,
-                issues: new List<SonarQubeIssue> { issue1, issue2, issue3, issue4, issue5, issue6, issue7, issue8, issue9, issue1 });
-
-            var issuesProvider = new SonarQubeIssuesProvider(mockSqService.Object, "sqKey",
-                mockTimerFactory.Object, testLogger);
+
+            SetupSolutionBinding(true, new List<SonarQubeIssue> { sonarQubeIssue });
+
+            var issuesProvider = new SonarQubeIssuesProvider(mockSqService.Object, "sqkey", mockTimerFactory.Object, testLogger);
             WaitForInitialFetchTaskToStart();
 
             VerifyServiceGetIssues(Times.Exactly(1)); // issues should be fetched on creation
 
-            // 1. Project id doesn't match -> no issues
-            var matches = issuesProvider.GetSuppressedIssues("unrecognisedProjectId", "folder1/file1");
-            matches.Should().BeEmpty();
-            VerifyServiceGetIssues(Times.Exactly(1), "sqKey"); // cached issues should be used
+            // Act
+            var matches = issuesProvider.GetSuppressedIssues("projectId", null);
 
-            // 2. File id doesn't match -> no issues
-            matches = issuesProvider.GetSuppressedIssues("projectID1", "folder1/filexxx");
-            matches.Should().BeEmpty();
+            // Assert
+            matches.Should().HaveCount(1);
```

Comment:
As discussed, this test doesn't prove that the product code is doing any filtering - there would need to be at least one other issue that is returned by the mocked server to test that filtering is happening.